tests/unit/s2n_x509_validator_test.c

```diff
@@ -1365,6 +1365,76 @@ int main(int argc, char **argv)
         s2n_x509_trust_store_wipe(&trust_store);
     };
 
+    /* test validator in safe mode, with default host name validator. Connection server matches the IPv6 address on the certificate. */
+    {
+        struct s2n_x509_trust_store trust_store;
```

Comment:
Make sure you're zero-initing everywhere. I know your next line inits it, but I just fixed a build issue related to zero-initing and I trust nothing.
```suggestion
    {
        struct s2n_x509_trust_store trust_store = { 0 };
```

Reply:
I copy/pasted this test from several other checks in this file so those should be fixed as well.

Reply:
You don't have to fix the other tests if you don't want to, but new code added should be correct.